torch/nn/modules/module.py

```diff
@@ -586,87 +613,162 @@ def state_dict(self, destination=None, prefix='', keep_vars=False):
+            version (int): The version number associated with the input
+                :attr:`local_state_dict`. If the input state dict was created
+                without a version number, this will be ``None``
+            strict (bool): Whether to strictly enforce that the keys in
+                :attr:`state_dict` match the keys returned by this module's
+                :meth:`~torch.nn.Module.state_dict` function.
         """
-        def submodule_key_mismatch(full_name, is_missing):
-            module = self
-            names = full_name.split(".")
-            for module_name in names[:-1]:
-                if module_name in module._modules:
-                    module = module._modules[module_name]
-                else:
-                    return
-            module._load_state_dict_key_mismatch(full_name, names[-1], is_missing)
+        own_local_state = self.state_dict(no_child=True)
 
         unexpected = []
-        own_state = self.state_dict()
-        for name, param in state_dict.items():
-            if name in own_state:
+        for name, param in local_state_dict.items():
+            if name in own_local_state:
                 if isinstance(param, Parameter):
                     # backwards compatibility for serialized parameters
                     param = param.data
                 try:
-                    own_state[name].copy_(param)
+                    own_local_state[name].copy_(param)
                 except Exception:
                     raise RuntimeError('While copying the parameter named {}, '
                                        'whose dimensions in the model are {} and '
                                        'whose dimensions in the checkpoint are {}.'
-                                       .format(name, own_state[name].size(), param.size()))
+                                       .format(name, own_local_state[name].size(), param.size()))
             elif strict:
                 unexpected.append(name)
+
         if strict:
-            missing = set(own_state.keys()) - set(state_dict.keys())
-            # pass the mismatch info to submodules so that they have a chance to
-            # raise a custom class-specific error
-            for name in unexpected:
-                submodule_key_mismatch(name, False)
-            for name in missing:
-                submodule_key_mismatch(name, True)
+            missing = set(own_local_state.keys()) - set(local_state_dict.keys())
             error_msg = ''
             if len(unexpected) > 0:
                 error_msg += 'Unexpected key(s) in state_dict: {}. '.format(
                     ', '.join('"{}"'.format(k) for k in unexpected))
             if len(missing) > 0:
                 error_msg += 'Missing key(s) in state_dict: {}. '.format(
-                    ', '.join('"{}"'.format(k) for k in unexpected))
+                    ', '.join('"{}"'.format(k) for k in missing))
```

Comment:
Do the keys have prefix removed at this point? If so, then this error will be very uninformative.